[BUG] panic: nil pointer dereference in isUserDefinedKeybinding when terminal sends OSC/CSI responses before model is initialized

# [BUG] panic: nil pointer dereference in `isUserDefinedKeybinding` when terminal sends OSC/CSI responses before model is initialized

**Describe the bug**

`gh dash` panics on launch with `runtime error: invalid memory address or nil pointer dereference` inside `(*Model).isUserDefinedKeybinding`. It happens when the host terminal proactively responds to bubbletea's startup queries (e.g. OSC 11 background-color, CSI cursor-position report) before the bubbletea program has finished setting up the model context. Those response bytes are parsed as `tea.KeyMsg` and dispatched to `Update`, which calls `isUserDefinedKeybinding(msg)` while `m.ctx` is still nil.

**Reproduction**

I see this consistently when launching `gh dash` from a [Wave Terminal](https://github.com/wavetermdev/waveterm) widget block (Wave eagerly emits color/cursor responses on tty open). The same panic has been reported elsewhere from xterm-style terminals that respond fast to the startup queries.

```
gh dash version dev
goos: darwin
goarch: arm64
module version: v4.22.0
module version: v4.23.2  (also reproduces)
```

**Stack trace (v4.22.0)**

```
panic: runtime error: invalid memory address or nil pointer dereference

github.com/dlvhdr/gh-dash/v4/internal/tui.(*Model).isUserDefinedKeybinding(...)
	internal/tui/ui.go:1416 +0x44
github.com/dlvhdr/gh-dash/v4/internal/tui.Model.Update(...)
	internal/tui/ui.go:208 +0xc1c
github.com/charmbracelet/bubbletea.(*Program).eventLoop(...)
	bubbletea@v1.3.5/tea.go:525
```

(v4.23.2 stack is identical with line numbers shifted: `ui.go:1459` and `ui.go:219`. Visible bytes preceding the panic include `^[]11;rgb:0000/0000/0000^[\` and `^[[1;1R`, confirming the rogue input is the OSC 11 / cursor-pos response.)

**Cause**

`isUserDefinedKeybinding` dereferences `m.ctx.Config.Keybindings.*` without checking whether `m.ctx` is nil. The `Update` loop can be entered with key messages before the context is wired up, when the terminal injects synthetic-looking input early.

**Suggested fix**

Guard the entry of `isUserDefinedKeybinding` with a nil-check on `m.ctx` and `m.ctx.Config`. PR with the 3-line guard incoming.

**Notes on AI usage** (per AI_POLICY.md)

I diagnosed and patched this with help from Claude Code. The fix is a single nil-guard at the top of the function — I have read it, can defend it, and built and verified the patched binary locally before opening this report.

--- a/internal/tui/ui.go
+++ b/internal/tui/ui.go
@@ -1630,6 +1630,9 @@ func (m *Model) switchSelectedView() tea.Cmd {
 }
 
 func (m *Model) isUserDefinedKeybinding(msg tea.KeyMsg) bool {
+	if m.ctx == nil || m.ctx.Config == nil {
+		return false
+	}
 	for _, keybinding := range m.ctx.Config.Keybindings.Universal {
 		if keybinding.Builtin == "" && keybinding.Key == msg.String() {
 			return true

